L1Trigger/L1THGCal/src/backend/HGCalMulticluster_SA.cc

```diff
@@ -0,0 +1,57 @@
+#include "L1Trigger/L1THGCal/interface/backend/HGCalMulticluster_SA.h"
+
+#include <cmath>
+
+using namespace l1t;
+
+HGCalMulticluster_SA::HGCalMulticluster_SA(const l1t::HGCalCluster_SA& tc, float fraction) {
+  HGCalMulticluster_SA();
+  addConstituent(tc, true, fraction);
+}
+
+void HGCalMulticluster_SA::addConstituent(const l1t::HGCalCluster_SA& tc, bool updateCentre, float fraction) {
+  double cMipt = tc.mipPt() * fraction;
+
+  // If no constituents, set seedMiptPt to cluster mipPt
+  if (constituents_.empty()) {
+    // seedMipPt_ = cMipPt;
+    if (!updateCentre) {
+      centre_x_ = tc.x();
+      centre_y_ = tc.y();
+      centre_z_ = tc.zside();
```

Comment:
Should be `tc.z()` instead of `zside()`. `zside` tells if it is positive or negative endcap